### ISSUE TITLE
Please make Sequel Ace remember which export type you last used. Or have a setting for the default.

### ISSUE THREAD

**Issue by coredumperror** [Sequel-Ace/Sequel-Ace#1622]
I never ever _ever_ **ever** want to export as CSV, but Sequel Ace almost ALWAYS defaults to that format when I use File -> Export. I switch to the SQL tab, and Sequel Ace remembers that for a little while, but eventually goes back to the CSV tab.

If "remembering the last one" is too onerous to implement, a setting in the preferences for "Default Export Type" would work, too.

**Comment by coredumperror** [comment]
That doesn't seem to be the case for me. I have not found any way to force the Export system to default to the SQL tab or to the CSV tab based on program state. It's either on CSV to start, or I've changed it to SQL and it hasn't yet changed back on its own. Even restarting the program doesn't reliably force it to change one way or the other.

**Comment by luis-** [comment]
From looking at the source code and running a quick test locally I see the following behavior:
* `Content` Tab select and 1 or more rows are loaded/displayed -> export defaults to `CSV` tab
* `Query` Tab select and 1 or more rows are loaded/displayed -> export defaults to `CSV` tab
* All other tabs or no rows displayed in `Content` | `Query` tab -> export defaults to `SQL` tab

**Comment by Jason-Morcos** [comment]
May I ask what's the big issue about defaulting to any one export type or the other? Isn't it just _one more click_ to switch between any of the export modes? Am I not understanding part of why this is a bigger issue?

**Comment by arlomedia** [comment]
I sometimes have to do two or four exports in a row, and if I have to switch from CSV to SQL every time, that "one more click" gets annoying. It's also breaks my mental flow because I expect it to be at SQL where I left it, and I sometimes get partway through configuring my export before I realize it's back to CSV. So "one more click" is less disruptive than breaking my concentration with thoughts like "oh, jeez, now I have to change that again, now where was I...."

I'd like to see either SQL be the default always, or remember the last format, which would be better for anyone who routinely exports in other formats.

**Comment by designconcepts** [comment]
+1 for a default export setting.

It seems that the structure view defaults to SQL and content view defaults to CSV. But I prefer the content view, so I would have to switch to SQL export every time. Right now this seems like a inconsist and confusing behaviour.

**Comment by adrianjean** [comment]
+1 for a default export setting.
I never use CSV, but can see it being a useful option. However, I'm often in content mode looking at the data before exporting and I have accidentally exported CSV files many times and not noticed until going back to the exported file — especially if moving quickly.

The ability to select a default export seems pretty logical. So the options could be "Auto" (current behaviour) or one of the export options available.

Not sure why the app is deciding for me what I want to export when.

Otherwise a really great app. Thanks for keeping it going!!

**Comment by arlomedia** [comment]
Instead of a new setting for the default export format, which someone would have to make an effort to find, I would vote for simply remembering the last-used export format. That would automatically give everyone what they probably want, and would be predictable.

It looks like the last-used export filename is already saved and loaded automatically, but the Path defaults to ~/Downloads. I'd like to see the format tab, the path and the filename all remember my last selection.

**Comment by rubenjleon** [comment]
Just for context, it’s literally used by millions all over the world including the U.S. government who offers exports in this format.  Major Databases allow data loaders based on this format and it’s older than XML and JSON and better for databases.  It’s been around since the 70s and I’ve personally have been using it since 2000…
Point is this is basic functionality that is expected to be there…. Not having it would greatly devalue the usefulness of the tool!


Sent from Yahoo Mail for iPhone


On Tuesday, October 7, 2025, 12:05 PM, Robert Rollins ***@***.***> wrote:

coredumperror left a comment (Sequel-Ace/Sequel-Ace#1622)
It's useless to me. I have literally never wanted to export a database as CSV.

**Comment by coredumperror** [comment]
I don't think you understand what our complaint here is. We just want CSV to not be the *default* option for database export, with no way to change it. Nobody's asking for the CSV export feature to be removed.

**Comment by rubenjleon** [comment]
I was simply addressing your comment about CSV format being useless!


Sent from Yahoo Mail for iPhone


On Tuesday, October 7, 2025, 3:10 PM, Robert Rollins ***@***.***> wrote:

coredumperror left a comment (Sequel-Ace/Sequel-Ace#1622)
I don't think you understand what our complaint here is. We just want CSV to not be the default option for database export, with no way to change it. Nobody's asking for the CSV export feature to be removed.
